File: 1-Basics/Lab1/Lab1-TypeScript/Person.ts
export class Person {
    private readonly _firstName: string;

    get firstName(): string {
        return this._firstName;
    }

    private readonly _lastName: string;

    get lastName(): string {
        return this._lastName;
    }

    constructor(firstName: string, lastName: string) {
        this._firstName = firstName;
        this._lastName = lastName;
    }

    public toString = () : string => {
        return this.firstName + " " + this.lastName;
    }
}

File: 1-Basics/Lab2/Lab2-C++/Person.cpp
#include <utility>
#include <string>

#include "Person.h"

Person::Person(std::string firstName, std::string lastName, Date dateOfBirth)
{
    _FirstName = std::move(firstName);
    _LastName = std::move(lastName);
    _DateOfBirth = dateOfBirth;
    getAllPerson()->push_back(this);
}

std::string Person::getAllPersonInfo()
{
    std::string result;
    for (auto person : _AllPerson)
    {
        result += person->ToString() + "\n";
    }
    return result;
}

std::string Person::ToString()
{
    return this->getFirstName() + " " + getLastName() + "; Дата рождения: " +
           getDateOfBirth().ToString();
}
